src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Flow.java

```diff
@@ -887,16 +912,23 @@ private List<Type> components(Type seltype) {
          * for the sealed supertype.
          */
         private List<PatternDescription> reduceBindingPatterns(Type selectorType, List<PatternDescription> patterns) {
+            Set<Symbol> existingBindings = patterns.stream()
```

Comment:
Playing some more - I found this example:

```
class Test {
    sealed interface I permits C, D { }
    non-sealed interface C extends I { }
    non-sealed interface D extends I { }

    interface F { }

    <Z extends I & F> int test(Z o) {
        return switch (o) {
            case C c -> 1;
            case D d -> 2;
        };
    }
}
```

Which compiles correctly, but it doesn't look exhaustive to me (because of F) ?

Reply:
Also, surprisingly, if I make C and D classes (instead of interfaces):
```
class Test {
    sealed interface I permits C, D { }
    final class C implements I { }
    final class D implements I { }

    interface F { }

    <Z extends I & F> int test(Z o) {
        return switch (o) {
            case C c -> 1;
            case D d -> 2;
        };
    }
}
```
I get errors like:

```
Foo.java:10: error: incompatible types: Z cannot be converted to Test.C
            case C c -> 1;
                 ^
  where Z is a type-variable:
    Z extends I,F declared in method <Z>test(Z)
```

Which seems odd?

Reply:
> Which compiles correctly, but it doesn't look exhaustive to me (because of F) ?

Can you expand a little more, because for me a Z either implement C or D or both C and D.

Reply:
> > Which compiles correctly, but it doesn't look exhaustive to me (because of F) ?
> 
> Can you expand a little more, because for me a Z either implement C or D or both C and D.

Please see my latest comment above. I was confused by the rules of coverage with intersection types. F is immaterial in this example - the only thing to show is that there is at least one of the component of the intersection (in this case I) that is fully covered.